modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/handlers/TxCleanupRecoveryRequestHandler.java

```diff
@@ -104,8 +107,11 @@ private void runPersistentStorageScan() {
                 });
             }
         } catch (IgniteInternalException e) {
-            String errorMessage = String.format("Failed to scan transaction state storage [commitPartition=%s].", replicationGroupId);
-            failureProcessor.process(new FailureContext(e, errorMessage));
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-25302 - remove this IF after proper stop is implemented.
+            if (!hasCause(e, TxStateStorageClosedException.class, TxStateStorageDestroyedException.class)) {
+                String errorMessage = String.format("Failed to scan transaction state storage [commitPartition=%s].", replicationGroupId);
+                failureProcessor.process(new FailureContext(e, errorMessage));
+            }
         }
 
         LOG.debug("Persistent storage scan finished [committed={}, aborted={}].", committedCount, abortedCount);
```

Comment:
This is weird that we will print this message in case the storage was closed

Reply:
Why? We did some work and report it, even if we were interrupted early